sysl2/sysl/ints.go

```diff
@@ -0,0 +1,215 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	sysl "github.com/anz-bank/sysl/src/proto"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type IntegrationDependency struct {
+	m            *sysl.Module
+	excludes     StrSet
+	passthroughs StrSet
+	drawableApps StrSet
+	apps         StrSet
+	deps         map[string]*AppDependency
+}
+
+func MakeIntegrationDependency(mod *sysl.Module, excludes, passthroughs, highlights StrSet) *IntegrationDependency {
+	return &IntegrationDependency{
+		m:            mod,
+		excludes:     excludes,
+		passthroughs: passthroughs,
+		drawableApps: highlights,
+		apps:         highlights.Clone(),
+		deps:         map[string]*AppDependency{},
+	}
+}
+
+func (v *IntegrationDependency) collectApplicationDependencies(apps StrSet) {
+	for app := range apps {
+		apps := v.m.GetApps()
```

Comment:
`apps := v.m.GetApps()` should be  outside the for loop, right?

Reply:
fixed